### PR TITLE
Do not allow docker on k8s 1.24+

### PR DESCRIPTION
**Pull request by olemarkus**
Ref:
* https://github.com/kubernetes/kubernetes/issues/106893
* https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2221-remove-dockershim/README.md

### CODE REVIEW

**Comment by hakman** [comment]
Should we wait for https://github.com/kubernetes/kubernetes/pull/106907 to merge first?

**Comment by olemarkus** [comment]
We can, but I don't think it really changes anything more on our end. Not that I can see at least.

**Comment by hakman** [comment]
Mean in case there is a decision to postpone it.

**Comment by olemarkus** [comment]
Think that is unlikely, but I have no problem with waiting a bit

File: pkg/apis/kops/validation/validation.go [hunk 1]
```diff
@@ -1271,11 +1271,15 @@ func validateCalicoEncapsulationMode(mode string, fldPath *field.Path) field.Err
...
-func validateContainerRuntime(runtime *string, fldPath *field.Path) field.ErrorList {
+func validateContainerRuntime(c *kops.Cluster, runtime string, fldPath *field.Path) field.ErrorList {
 	valid := []string{"containerd", "docker"}
 
 	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, IsValidValue(fldPath, runtime, valid)...)
+	allErrs = append(allErrs, IsValidValue(fldPath, &runtime, valid)...)
+
+	if runtime == "docker" && c.IsKubernetesGTE("1.24") {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "docker is not supported by Kubernetes 1.24 and above"))
```
Comment (hakman) [inline pkg/apis/kops/validation/validation.go:1281]:
```suggestion
		allErrs = append(allErrs, field.Forbidden(fldPath, "Docker CRI support was removed in Kubernetes 1.24: https://kubernetes.io/blog/2020/12/02/dockershim-faq"))
```

**Review by hakman — CHANGES_REQUESTED** [review]
Considering https://github.com/kubernetes/kubernetes/pull/97252 was merged, I think it's safe to merge this too.

**Review by hakman — APPROVED** [review]

**Pull request merged**